lib/Core/MemoryManager.cpp

```diff
@@ -40,7 +98,31 @@ MemoryObject *MemoryManager::allocate(uint64_t size, bool isLocal,
     klee_warning_once(0, "Large alloc: %lu bytes.  KLEE may run out of memory.",
                       size);
 
+  // Return NULL if size is zero, this is equal to error during allocation
+  if (NullOnZeroMalloc && size == 0)
+    return 0;
+
   uint64_t address = 0;
+  if (DeterministicAllocation) {
+    if ((alignment & -alignment) != alignment) {
+      klee_warning("Only alignment of power of two is supported");
+      return 0;
+    }
+
+    address = ((uint64_t)nextFreeSlot + alignment - 1) & -alignment;
```

Comment:
I don't feel great about you negating ``alignment`` here which is of type ``size_t`` which is an unsigned type. Presumably this will promote it to a signed type and then I'm unsure how that interacts with``&`` (not sure if lhs gets promoted or not). You are also assuming two's complement behaviour here which I don't think is guaranteed by the C++ standard. I think it would be better to stay with unsigned types here so you should replace

``-alignment`` with ``(~alignment + 1)``.